src/Adapter/Profile/Permission/QueryHandler/GetPermissionsForConfigurationHandler.php

```diff
@@ -158,19 +159,29 @@ private function getTabsForPermissionsConfiguration()
     /**
      * @param array $tabs
      * @param int $parentId
+     * @param int $nestingLevel
      *
      * @return array
      */
-    private function buildTabsTree(array &$tabs, $parentId = 0)
+    private function buildTabsTree(array &$tabs, $parentId = 0, $nestingLevel = 1)
     {
+        $maxNestingLevel = 32;
+        if ($nestingLevel > $maxNestingLevel) {
+            throw new RuntimeException(sprintf(
+                'Maximum nesting level of "%s" reached in "%s"'
+                , $maxNestingLevel,
+                __METHOD__
+            ));
+        }
+
         $children = [];
 
         foreach ($tabs as &$tab) {
             $id = $tab['id'];
 
             if ((int) $tab['id_parent'] === (int) $parentId) {
                 $children[$id] = $tab;
-                $children[$id]['children'] = $this->buildTabsTree($tabs, $id);
+                $children[$id]['children'] = $this->buildTabsTree($tabs, $id, $nestingLevel++);
```

Comment:
You must use `$nestingLevel + 1` otherwise your counter increases at each loop and you will reach the limit really quickly

Reply:
I guess you are right, but this is also tricky. Let' say we call `$this->buildTabsTree()`, it may call itself 20 more times, but now we only care about how deep it will go. So it might be nesting level of 4, but repeated 20 times, not sure if that's an issue. :thinking:

Reply:
That's not an issue, the purpose of this check is to avoid infinite loop via recursive calls, the counter must increase when you perform a recursive call hence the `$nestingLevel + 1`
It's not a problem if you have 40 children in a tab, we should be able to deal with it